t/FakeModuleStaticInjection.pm

```diff
@@ -47,4 +49,12 @@ sub useImportedStaticFunction {
     return $PreFix . ': '.ReturnHelloWorld(@_);
 
 }
+sub dependency {
+    my ($self, $arg) = @_;
```

Comment:
No constructor, No self! ;-)

Reply:
I was following your lead for consistency with the rest of the file. But I agree, if the module does not represent a class, then there really shouldn't be any shifting for $self within each function. Furthermore, calls to that function should be of the form `FakeModuleStaticInjection::functionName`, not `FakeModuleStaticInjection->functionName`. I don't particularly care for this invocation flexibility in Perl. I will see about cleaning this up.

Reply:
I added some TODO statements as a reminder. I do not have time to get into it now.